runtime/ms-rest-azure/index.d.ts

```diff
@@ -241,15 +241,18 @@ export class BaseResource {
  *
```

Comment:
keep `index.d.ts` right next to `index.js` (whichever is the entry point) inside `lib`. See [TS Handbook](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) for reference.

The convention is that the TS entry point can be found in the same path as the JS entry point and thats likely where `tsc.exe` looks if we didn't specify a `typings` in `package.json`. Why do the extra effort to break it 😃